assets/javascripts/discourse/components/canned-reply.js.es6

```diff
@@ -0,0 +1,71 @@
+import showModal from 'discourse/lib/show-modal';
+import { ajax } from 'discourse/lib/ajax';
+import { popupAjaxError } from 'discourse/lib/ajax-error';
+
+export default Ember.Component.extend({
+
+  /**
+   * Whether the content is shown or the excerpt.
+   *
+   * @type {Boolean}
+   */
+  isOpen: false,
+
+  actions: {
+
+    /**
+     * Opens the content and hides the excerpt.
+     */
+    open: function() {
+      this.set('isOpen', true);
+    },
+
+    /**
+     * Hides the content and shows the excerpt.
+     */
+    close: function() {
+      this.set('isOpen', false);
+    },
+
+    /**
+     * Applies the canned reply.
+     */
+    apply: function() {
+      // TODO: This is ugly. There _must_ be another way.
+      // TODO: This code is also duplicated (see controller).
+      const composer = Discourse.__container__.lookup('controller:composer');
```

Comment:
Is there any way to use the plugin API for this? 

https://github.com/discourse/discourse/blob/master/app/assets/javascripts/discourse/lib/plugin-api.js.es6#L55